### ISSUE TITLE
Add `shouldRescheduleFor` method (similar to `shouldInterrupt`)

### ISSUE THREAD

**Issue by cristiancalara** [thomasjohnkane/snooze#102]
I'm creating a zip in AWS that might take as long as 15m. I'd like to send the notification as soon as possible, so ideally I'd check every minute, and then if it's ready I send the notification, if not, I reschedule.

I've done this by extending the custom model and overwriting the send method. 
```php
public function send(): void
{
    // ...
    if ($this->shouldInterrupt($notification, $notifiable)) {
        $this->cancel();
        event(new NotificationInterrupted($this));

        return;
    }

    if($sendAt = $this->shouldRescheduleFor($notification, $notifiable)) {
        $this->reschedule($sendAt);

        return;
    }
    // ...

}

public function shouldRescheduleFor(?object $notification = null, ?object $notifiable = null):
\DateTimeInterface|string|null
{
    if (! $notification) {
        $notification = $this->serializer->unserialize($this->notification);
    }

    if (! $notifiable) {
        $notifiable = $this->serializer->unserialize($this->target);
    }

    if (method_exists($notification, 'shouldRescheduleFor')) {
        return $notification->shouldRescheduleFor($notifiable);
    }

    return null;
}
```

Created the issue mainly to ask if you think this might be a useful feature for the package. If that's the case, I can create a PR for this. 

Thanks for the package! :)

**Comment by atymic** [comment]
Feel free to PR, sounds great!

**Comment by atymic** [comment]
Closed in #104  :D
